frontend/components/ticket/Ticket.vue

```diff
@@ -122,220 +122,205 @@
+
+  import Mention from "../elements/mention/Mention";
+  import Tab from "../elements/Tab"
+  import ExternalTab from "./ExternalTab";
+  import InternalTab from "./InternalTab";
+  import Card from '../elements/card/Card'
+  import AttachmentsTab from "./AttachmentsTab";
+  import Avatars from "../elements/Avatars";
+  import EditShareWith from "./EditShareWith";
+  import UserDropdown from "../elements/dropdown/UserDropdown";
+  import LabelDropdown from "../elements/dropdown/LabelDropdown";
+  import RecentQuestions from "./RecentQuestions";
+  import {calendarDate} from "../../utils";
+
+  export default {
+    components: {
+      EditShareWith,
+      Avatars,
+      AttachmentsTab,
+      InternalTab,
+      ExternalTab,
+      Mention,
+      UserDropdown,
+      LabelDropdown,
+      Avatar,
+      Comment,
+      VueTribute,
+      RecentQuestions,
+      Tab,
+      Card
     },
-    onReplyPost: function () {
-      axios.get("/api" + window.location.pathname + "/replies").then(response => {
-        this.replies = response.data;
-
-        return axios.get("/api" + window.location.pathname + "/events")
-      }).then(response => {
-        this.events = response.data;
-      });
-
-
-      axios.get("/api" + window.location.pathname).then(response => {
-        this.ticket = response.data;
-      });
+    data() {
+      return {
+        inbox: null,
+        ticket: null,
+        replies: [],
+        labels: [],
+        comments: [],
+        staff: [],
+        events: [],
+        activeTab: 'external',
+        user: {},
+        role: "",
+        shared_with: [],
+        errors: [],
+        status: {
+          PNDG: 'Pending',
+          ASGD: 'Assigned',
+          ANSD: 'Awaiting response',
+          CLSD: 'Closed'
+        }
+      }
     },
-    onCommentPost: function () {
-      axios.get("/api" + window.location.pathname + "/comments").then(response => {
-        this.comments = response.data;
-      });
-
+    created() {
       axios.get("/api" + window.location.pathname).then(response => {
-        this.ticket = response.data;
+        this.ticket = response.data.ticket;
+        this.labels = response.data.ticket.labels;
+        this.user = response.data.me;
+        this.role = response.data.role;
+        this.inbox = response.data.inbox;
+
+        axios.defaults.xsrfCookieName = 'csrftoken';
+        axios.defaults.xsrfHeaderName = "X-CSRFTOKEN";
+
+        if (this.isStaff()) {
+          axios.get("/api" + window.location.pathname + "/comments").then(response => {
+            this.comments = response.data;
+          });
+
+          axios.get("/api/inboxes/" + this.ticket.inbox + "/staff").then(response => {
+            this.staff = response.data;
+          });
+
+          axios.get("/api/inboxes/" + this.ticket.inbox + "/users/" + this.ticket.author.id + "/roles").then(response => {
+            this.$set(this.ticket.author, 'role', response.data)
+          })
```

Comment:
Kunnen we deze niet ook in de ticket api stopen, als de request.user in Django een staff member is. Vervolgens hier dan checken of de data er is en op basis daarvan het laten zien of niet. Scheelt waar requests

Reply:
Ook worden de /events niet meer in geladen volgens mij